app/assets/javascripts/discourse/app/components/post-list/item/index.gjs

```diff
@@ -23,44 +32,136 @@ export default class PostListItem extends Component {
     }
   }
 
+  get hiddenClass() {
+    return (
+      this.args.post.hidden && !(this.currentUser && this.currentUser.staff)
+    );
+  }
+
+  get deletedClass() {
+    return this.args.post.deleted ? "deleted" : "";
+  }
+
+  get user() {
+    return {
+      id: this.args.post.user_id,
+      name: this.args.post.name,
+      username: this.args.post.draft_username || this.args.post.username,
+      avatar_template: this.args.post.avatar_template,
+      title: this.args.post.user_title,
+      primary_group_name: this.args.post.primary_group_name,
+    };
+  }
+
+  get postId() {
+    return this.args.post.id || this.args.post.post_id;
+  }
+
+  @action
+  buildOneboxes(element) {
+    schedule("afterRender", () => {
+      loadOneboxes(
+        element,
+        ajax,
+        this.args.post.topic_id,
+        this.args.post.category_id,
+        this.siteSettings.max_oneboxes_per_post,
+        true
+      );
+    });
+  }
+
+  @action
+  resolveShortUrls(element) {
+    resolveAllShortUrls(
+      ajax,
+      this.siteSettings,
+      element,
+      this.args.shortUrlOpts
+    );
+  }
```

Comment:
Similar to the onebox thing, this could lead to a cascade of ajax requests & rate limiting. The server-side cooking should've already populated these  🤔